### PR TITLE
Added wikis API

### LINKED ISSUE #437: Wikis
No methods are implemented for managing Wikis: https://docs.gitlab.com/ce/api/wikis.html

### PR DESCRIPTION
**Pull request by akkee**
Wikis API

`Gitlab.wikis(project)`
`Gitlab.wiki(project, slug)`
`Gitlab.create_wiki(project, title, content, options = {})`
`Gitlab.update_wiki(project, slug, options)`
`Gitlab.delete_wiki(project, slug)`

Closes #437

### CODE REVIEW

File: lib/gitlab/client/wikis.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    #   Gitlab.create_wiki(3, 'Some Content', 'Some Title')
+    #   Gitlab.create_wiki(3, 'Some Content', 'Some Title', { format: 'rdoc' })
+    #
+    # @param  [Integer, String] project The ID or name of a project.
+    # @param  [String] content The content of the wiki page.
+    # @param  [String] title The title of the wiki page.
+    # @param  [Hash] options A customizable set of options.
+    # @option options [String] format (optional) The format of the wiki page. Available formats are: markdown (default), rdoc, and asciidoc.
+    # @return [Gitlab::ObjectifiedHash] Information about created wiki page.
+    def create_wiki(project, content, title, options = {})
```
Comment (NARKOZ) [inline lib/gitlab/client/wikis.rb]: Can we swap title and content so a title gets passed first?

```suggestion
    def create_wiki(project, title, content, options = {})
```

Comment (akkee) [reply lib/gitlab/client/wikis.rb]: Agreed, done

**Comment by NARKOZ** [comment]
Thank you ❤️

File: lib/gitlab/client/wikis.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    def create_wiki(project, title, content, options = {})
+      body = { content: content, title: title }.merge(options)
+      post("/projects/#{url_encode project}/wikis", body: body)
+    end
+
+    # Updates an existing wiki page. At least one parameter is required to update the wiki page.
+    #
+    # @example
+    #   Gitlab.update_wiki(6, 'home', { title: 'New title' })
+    #   Gitlab.update_wiki(6, 'home', { title: 'New title', message: 'New Message', format: 'rdoc' })
```
Comment (connorshea) [inline lib/gitlab/client/wikis.rb]: `message`?

File: lib/gitlab/client/wikis.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    # @return [Gitlab::ObjectifiedHash]
+    def wiki(project, slug)
+      get("/projects/#{url_encode project}/wikis/#{slug}")
+    end
+
+    # Creates a new wiki page for the given repository with the given title, slug, and content.
+    #
+    # @example
+    #   Gitlab.create_wiki(3, 'Some Content', 'Some Title')
+    #   Gitlab.create_wiki(3, 'Some Content', 'Some Title', { format: 'rdoc' })
```
Comment (connorshea) [inline lib/gitlab/client/wikis.rb]: The content and title need to be switched here, and in the parameters below

File: lib/gitlab/client/wikis.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  module Wikis
+    # Get all wiki pages for a given project.
+    #
+    # @example
+    #   Gitlab.wikis(3)
+    #   Gitlab.wikis(3, {with_content: 'Some wiki content'})
+    #
+    # @param  [Integer, String] project The ID or name of a project.
+    # @param  [Hash] options A customizable set of options.
+    # @option options [String] :with_content(optional) Include pages content
```
Comment (connorshea) [inline lib/gitlab/client/wikis.rb]: Is `:with_content` supposed to be a symbol?

Comment (connorshea) [reply lib/gitlab/client/wikis.rb]: It looks like it is, but the other options in this file aren't symbols, that should probably be fixed :)

File: lib/gitlab/client/wikis.rb (hunk 2 above)
Comment (NARKOZ) [reply lib/gitlab/client/wikis.rb]: PR is welcome

**Pull request merged**